### ISSUE TITLE
Picker window remains open after selection

### ISSUE THREAD

**Issue by jssee** [srstevenson/vim-picker#37]
currently Picker uses `bdelete` after a selection is made, however this doesnt ensure that the :terminal window closes. Sometimes, the window will stay open at the bottom, and next time the picker command is invoked, it stacks below the previous picker window. Is there a way to use both `bdelete` and `close` so that the window closes and the buffer is deleted after selection?

**Comment by boxofrox** [comment]
I've never had this issue with vim-picker in Neovim.  Do you observe this behavior in Neovim, Vim, or both?  Also mention which version(s) your using.

There's also the possibility that there's a conflict with another plugin you're using.  Can you generate a minimal .vimrc that reproduces the issue?  For example:

```
$ mkdir -p ~/vim-picker-test
$ cd ~/vim-picker-test
$ git clone https://github.com/srstevenson/vim-picker.git
$ cat > init.vim <<'EOF'
set rtp^=./vim-picker
runtime! plugin/**/*.vim

nmap <leader>pb <Plug>PickerBuffer
EOF
$ nvim -u ./init.vim --noplugin
```

**Comment by jssee** [comment]
After some more poking around, it seems to only happen when using Picker from A “file explorer” buffer. I’ve also seen it happen with vaffle.vim and also plain netrw. If I start him from a specific file, things seem to work as expected

**Comment by srstevenson** [comment]
Thanks for the follow up @jssee: I can reproduce this by opening Netrw with `:Explore` in a newly launched Neovim instance, and opening vim-picker. The issue is that the Netrw buffer is marked as unlisted: in the case that there are no other listed buffers, `bdelete!` doesn't close the window. You can see this without vim-picker with the following steps:

1. open a new instance of Neovim;
2. open Netrw in a new window with `:Sexplore`;
3. move to the original window with <kbd>Ctrl-w</kbd> <kbd>p</kbd>;
4. run `:bdelete!` and observe that the window isn't closed.

I've opened #39 with the minimum viable resolution of calling `:close!` instead of `:bdelete!`. This fixes the problem in some brief testing I've done with Neovim.

**Comment by jssee** [comment]
Hey @srstevenson , thanks for responding.

Originally when I started to encounter this behavior, I came to the same conclusion, which is to change the `bdelete` command for `close`. I experimented with it but realized that although doing this ensures that the window closes, you end up opening a new buffer every time Picker is invoked. which is a lot it turns out if youre using all of its feature. This can be seen by running the `:buffers` command after switching to a few different files using picker.

This is why i was originally thinking the best solution would be to run both `bdelete` and `close` but idk enough vimscript to know if thats possible.

**Comment by srstevenson** [comment]
The behaviour you describe was what I was expecting when I wrote #39, so at first I executed `:close` with `bufhidden` set to `delete` so that hidden buffers are deleted:

```vim
let l:bufhidden = &bufhidden
set budhidden=delete
close!
let &bufhidden = l:bufhidden
```
However out of curiosity I tried with just `close!`, and in both Neovim 0.3.3 and Vim 8.1 that left no lingering vim-picker buffers when I tried it. Do the vim-picker buffers get left around using a minimal vimrc like the one @boxofrox posted above, or just when using your normal configuration? If the latter we can try bisecting your vimrc to determine if some other option is causing us to see the different behaviour.
